File: src/main/java/io/github/udonabe/donabe/compile/code/ByteCode.java
package io.github.udonabe.donabe.compile.code;

import io.github.udonabe.donabe.compile.code.section.ConstantPoolSection;
import io.github.udonabe.donabe.compile.code.section.GlobalIdentifiersSection;
import io.github.udonabe.donabe.compile.code.section.InitializationCodeSection;
import io.github.udonabe.donabe.compile.code.section.Section;
import java.util.HashSet;
import java.util.Set;

public record ByteCode(GlobalIdentifiersSection globalIdentifiersSection,
        ConstantPoolSection constantPoolSection,
        InitializationCodeSection initializationCodeSection) {    
}

File: Donabe-VM/src/bytecode.rs
use crate::bytecode::section::{InitializationCodeSection, ConstantPoolSection, GlobalIdentifiersSection};

pub mod section;

#[derive(Debug)]
pub struct ByteCode {
    pub constant_pool: ConstantPoolSection,
    pub identifiers: GlobalIdentifiersSection,
    pub code_section: InitializationCodeSection,
}